src/components/contents/UserProfile.js

```diff
@@ -0,0 +1,139 @@
+import React, {PropTypes} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import userPlaceholder from '../../res/images/profile/placeholder-user.png';
+import smilies from '../../res/images/common/smilies.png';
+import TextInput from '../common/TextInput';
+import * as sessionActions from '../../actions/sessionActions';
+import * as alertActions from '../../actions/alertActions';
+import { ALERT_GOALS  } from '../../enums/enums';
+import { validateUserProfile } from '../../utils/ValidationHelper';
+
+
+class UserProfile extends React.Component{
+  constructor(props, context){
+    super(props, context);
+
+    this.onSettingsSubmit = this.onSettingsSubmit.bind(this);
+    this.onFieldChange = this.onFieldChange.bind(this);
+    this.state = {
+      email: this.props.session.user.email,
+      currentPassword: '',
+      newPassword: '',
+      confirmPassword: ''
+    };
+  }
+
+  onSettingsSubmit(e){
+    e.preventDefault();
+
+    const newSettings = {
+      user: {
+        prev: this.state.currentPassword,
```

Comment:
use destructuring object for each prop: `const { currentPassword, newPassword, etc } = this.state;`